tests/cli-v2.rs

```diff
@@ -808,6 +809,61 @@ fn remove_target_missing_update_hash() {
     });
 }
 
+// Issue #1777
+#[test]
+fn warn_about_and_remove_stray_hash() {
+    setup(&|config| {
+        let mut hash_path = config.rustupdir.join("update-hashes");
+        fs::create_dir_all(&hash_path).expect("Unable to make the update-hashes directory");
+
+        hash_path.push(for_host!("nightly-{}"));
+        println!("Update-hash path: {}", hash_path.display());
+
+        let mut file = fs::File::create(&hash_path).expect("Unable to open update-hash file");
+        file.write_all(b"LEGITHASH")
+            .expect("Unable to write update-hash");
+        drop(file);
+
+        expect_ok_ex(
```

Comment:
The use of `expect_ok_ex` means you are very brittle against the exact output of the tool.

Instead I'd look to use `expect_stdout_ok` to check for the warning, and then possibly `expect_ok` to check that `rustc --version` works for the toolchain.  That way you can demonstrate that (a) the warning is emitted and (b) the toolchain got installed, without being brittle against the exact output.